src/aosm/azext_aosm/deploy/pre_deploy.py

```diff
@@ -131,6 +130,28 @@ def ensure_config_publisher_exists(self) -> None:
             location=self.config.location,
         )
 
+    def ensure_config_source_registry_exists(self) -> None:
+        """
+        Ensures that the source registry exists
+
+        Finds the parameters from self.config
+        """
+
+        logger.info(
+            "Check if the source registry %s exists",
+            self.config.source_registry_id,
+        )
+
+        source_registry_name = self.config.source_registry_id.split("/")[-1]
+        source_registry_resource_group_name = self.config.source_registry_id.split("/")[
```

Comment:
This looks like you are making some assumption based on the form of the id.   Can you write down what those are in a comment?